### PR TITLE
Change write to return a Completable

### LINKED ISSUE #16: The write method should throw when writing fails
RSA encryption only allows 256 bytes to be encrypted, so if you try to write a large value then Whorlwind will silently fail with a log in the following format:

```
Failed to write value for <name>
javax.crypto.IllegalBlockSizeException: input must be under 256 bytes
```

Of course there are other reasons why this might fail too, but this one is probably one that people will run into often. It would be better to be able to handle this as an error. An alternative solution would be to have this method return a boolean that returns true only when the write was successful.

### PR DESCRIPTION
**Pull request by mattprecious**
Fixes #16 in the process...

cc: @oldergod

### CODE REVIEW

File: whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java [hunk 1]
```diff
@@ -90,26 +92,26 @@ void checkCanStoreSecurely() {
     }
   }
 
-  @Override public void write(@NonNull String name, @Nullable ByteString value) {
-    checkCanStoreSecurely();
+  @Override public Completable write(@NonNull final String name, @Nullable final ByteString value) {
+    return Completable.fromAction(new Action() {
+      @Override public void run() throws Exception {
+        checkCanStoreSecurely();
```
Comment (mattprecious) [inline whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java]: Actually, I might want to pull this outside of the action... it's outside the subscription in `read`

Comment (JakeWharton) [reply whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java]: Maybe. I would probably argue the opposite though. Creating the observable isn't actually doing anything.

File: whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java [hunk 2]
```diff
@@ -90,26 +92,26 @@ void checkCanStoreSecurely() {
     }
   }
 
-  @Override public void write(@NonNull String name, @Nullable ByteString value) {
-    checkCanStoreSecurely();
+  @Override public Completable write(@NonNull final String name, @Nullable final ByteString value) {
+    return Completable.fromAction(new Action() {
+      @Override public void run() throws Exception {
```
Comment (oldergod) [inline whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java]: nit: you can remove the `throws Exception`

File: whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java (hunk 1 above)
Comment (JakeWharton) [reply whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java]: The doc of `canStoreSecurely` makes it seem like the check should be outside.

File: whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java [hunk 3]
```diff
@@ -90,26 +92,26 @@ void checkCanStoreSecurely() {
...
+        checkCanStoreSecurely();
 
-    synchronized (dataLock) {
-      if (value == null) {
-        storage.remove(name);
-        return;
-      }
+        synchronized (dataLock) {
```
Comment (oldergod) [inline whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java]: This and others access private members; you could private-package those fields to reduce method counts

File: whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java (hunk 2 above)
Comment (mattprecious) [reply whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java]: I can't, actually. Compilation error.

File: whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java (hunk 1 above)
Comment (mattprecious) [reply whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java]: I can update the docs. `read` throws both outside and inside the observable... I can remove the one outside.

Comment (JakeWharton) [reply whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java]: Let's do that. Observable factory methods should only do argument validation.

File: whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java (hunk 3 above)
Comment (mattprecious) [reply whorlwind/src/main/java/com/squareup/whorlwind/RealWhorlwind.java]: Done

File: whorlwind/src/main/java/com/squareup/whorlwind/NullWhorlwind.java [hunk 4]
```diff
@@ -23,7 +24,7 @@
     return false;
   }
 
-  @Override public void write(String name, ByteString value) {
+  @Override public Completable write(String name, ByteString value) {
     throw new UnsupportedOperationException();
```
Comment (oldergod) [inline whorlwind/src/main/java/com/squareup/whorlwind/NullWhorlwind.java]: How about returning `Completable.error()`, same for `read()` ?

**Pull request merged**